File: esbmc.cpp/validador_titulo/validador_repetidos_consecutivos_titulo.cpp
#define MAX_LEN_TITLE 21

using namespace std;

bool validate_consecutive_repeated(const char title[MAX_LEN_TITLE], int len) {
    for (int i = 0; i < len - 1; i++) {
        __ESBMC_assert(title[i] != title[i + 1], "O título não deve conter caracteres repetidos consecutivos.");
    }
    return true;
}

int main() {
    char title[MAX_LEN_TITLE];
    int len;

    __ESBMC_assume(len >= 1 && len <= 20);

    validate_consecutive_repeated(title, len);

    return 0;
}

File: esbmc.py/Testes02/validador_titulo/validador_repetidos_consecutivos_titulo.py
MAX_LEN_TITLE: int = 21

def validate_consecutive_repeated(title: str, length: int) -> bool:
    i: int = 0
    while i < length - 1:
        assert title[i] != title[i + 1], \
            "O título não deve conter caracteres repetidos consecutivos."
        i += 1
    return True


def main() -> None:
    title: str = input()      
    length: int = len(title)

    assert 1 <= length <= 20  

    validate_consecutive_repeated(title, length)


main()
